### ISSUE TITLE
Flaky test: TestHiveDistributedQueriesWithOptimizedRepartitioning.testRunawayRegexAnalyzerTimeout

### ISSUE THREAD

**Issue by mbasmanova** [prestodb/presto#20764]
https://github.com/prestodb/presto/actions/runs/6086875767/job/16514229290?pr=20763

```
Error:  Failures: 
Error:    TestHiveDistributedQueriesWithOptimizedRepartitioning.testRunawayRegexAnalyzerTimeout » Test 
The exception was thrown with the wrong message: expected "Regexp matching interrupted" but got "The query optimizer exceeded the timeout of 1.00s."
[INFO] 
Error:  Tests run: 496, Failures: 1, Errors: 0, Skipped: 1
```

CC: @tdcmeehan @aditi-pandit @rschlussel

**Comment by Akanksha-kedia** [comment]
@tdcmeehan Can you help to assign to me.

**Comment by ClarenceThreepwood** [comment]
@rschlussel - This looks like a duplicate of #20411, and I posted a fix for that in #22286

**Comment by tdcmeehan** [comment]
Duplicate
